### PR TITLE
posix arch: support building on aarch64 / allow host-specific cmake includes

### LINKED ISSUE #24842: Support Building on Aarch64
**Is your enhancement proposal related to a problem? Please describe.**
Building on Aarch64 is pretty easy for native posix tests

**Describe the solution you'd like**
For the build to not try to pass `-m32` to the compiler, unconditionally.

**Describe alternatives you've considered**
Submitting a patch.

**Additional context**
Contents of `~/.zephyrrc`:
```bash
export BOARD=native_posix
export ZEPHYR_PROJECT=tests/net/socket/socketpair
```

Before:
```
2/126] Building C object zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj
FAILED: zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj 
ccache /usr/bin/gcc -DBUILD_VERSION=zephyr-v2.2.0-1657-g0f37c1e23a9c -DKERNEL -D_FORTIFY_SOURCE=2 -D_POSIX_C_SOURCE=200809 -D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED -D__ZEPHYR__=1 -I../kernel/include -I../arch/posix/include -I../include -Izephyr/include/generated -I../soc/posix/inf_clock -I../boards/posix/native_posix -I../subsys/net/lib/sockets/. -I../subsys/testsuite/include -I../subsys/testsuite/ztest/include -Os -imacros/home/chrisfriedt/workspace/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/home/chrisfriedt/workspace/zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/chrisfriedt/workspace/zephyrproject/zephyr/tests/net/socket/socketpair=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/chrisfriedt/workspace/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/chrisfriedt/workspace/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -m32 -include /home/chrisfriedt/workspace/zephyrproject/zephyr/arch/posix/include/posix_cheats.h -fno-freestanding -std=c11 -MD -MT zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj   -c /home/chrisfriedt/workspace/zephyrproject/zephyr/arch/posix/core/offsets/offsets.c
gcc: error: unrecognized command line option ‘-m32’
ninja: build stopped: subcommand failed.
```

After:
```
$ west build $ZEPHYR_PROJECT -t run
-- west build: generating a build system
Including boilerplate (Zephyr base (cached)): /home/chrisfriedt/workspace/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/chrisfriedt/workspace/zephyrproject/zephyr/tests/net/socket/socketpair
-- Zephyr version: 2.2.99 (/home/chrisfriedt/workspace/zephyrproject/zephyr)
-- Board: native_posix
-- Found west: /home/chrisfriedt/.local/bin/west (found suitable version "0.7.2", minimum required is "0.7.1")
-- Found toolchain: host (gcc/ld)
-- Found BOARD.dts: /home/chrisfriedt/workspace/zephyrproject/zephyr/boards/posix/native_posix/native_posix.dts
-- Generated zephyr.dts: /home/chrisfriedt/workspace/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/chrisfriedt/workspace/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
Parsing /home/chrisfriedt/workspace/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/chrisfriedt/workspace/zephyrproject/zephyr/build/zephyr/.config'
No change to configuration in '/home/chrisfriedt/workspace/zephyrproject/zephyr/build/zephyr/.config'
No change to Kconfig header in '/home/chrisfriedt/workspace/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- Cache files will be written to: /home/chrisfriedt/.cache/zephyr
-- Configuring done
-- Generating done
-- Build files have been written to: /home/chrisfriedt/workspace/zephyrproject/zephyr/build
-- west build: running target run
...
*** Booting Zephyr OS build zephyr-v2.2.0-1657-g0f37c1e23a9c  ***
```

### PR DESCRIPTION
**Pull request by cfriedt**
This change enables specific compiler and linker options to be used in the case that an `arch/posix/os.arch.cmake` file exists.

Note: os and arch in the above case are evaluations of [CMAKE_HOST_SYSTEM_NAME](https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_NAME.html) and [CMAKE_HOST_SYSTEM_PROCESSOR](https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html).

Otherwise, the existing "generic" compiler and linker flags in `arch/posix/CMakeLists.txt` are used.

Additional flags and checks are provided in `arch/posix/Linux.aarch64.cmake`.
    
Added `scripts/user_wordsize.py` to detect if userspace is 64-bit or 32-bit, which should be consistent with the value of `CONFIG_64BIT` for Aarch64 on Linux.
    
Fixes #24842

Output on Aarch64 with 64-bit userspace

Before (`-m32` is an x86 option for gcc):
```bash
gcc: error: unrecognized command line option ‘-m32’
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /home/chrisfriedt/workspace/zephyrproject/zephyr/build --target run
```
After (notice informative message):
```
CMake Error at ../../../../arch/posix/Linux.aarch64.cmake:27 (message):
  CONFIG_64BIT=n but this Aarch64 machine has a 64-bit userspace.

  If using cmake, consider using '-DBOARD=native_posix_64'

  If using west, consider using '-b native_posix_64'
```

After (with flags):
```
west build -b native_posix_64 tests/kernel/pipe/pipe_api -t run
...
Test suite pipe_api succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

### CODE REVIEW

**Comment by cfriedt** [comment]
Clearly, for prime-time, you guys will want to package a toolchain with the SDK, tools, etc.
Maybe soon you'll be running your builds on #Nuvia silicon ;-)

File: arch/posix/CMakeLists.txt
Comment (stephanosio) [inline arch/posix/CMakeLists.txt:9]: CMake supports `elseif(...)`

File: arch/posix/CMakeLists.txt
Comment (stephanosio) [inline arch/posix/CMakeLists.txt:10]: Nits: These should be indented with 2 spaces.

File: arch/posix/CMakeLists.txt
Comment (aescolar) [inline arch/posix/CMakeLists.txt:3]: Would you be so kind as to add a comment around here describing what this is doing?
I'm hoping for something like:
`# Detect the host architecture, so we can set the appropriate compile options flags depending also on the selected native_posix word size.
Otherwise this change adds to the mystic feel of our cmake files

File: arch/posix/CMakeLists.txt
Comment (aescolar) [inline arch/posix/CMakeLists.txt:6]: (Disclaimer I do not have any experience compiling in an ARM environment)
* Maybe a stupid question, but, do we know in general if the ABI defaults to lp64 in aarch64? (and/or would it need to be specified explicitly?)
* I guess that with this code as is only native_posix_64 compiles as expected?
  * And if that is the case I guess a `if (CONFIG_64BIT)` before this line and an else below with an error saying something like "only native_posix_64 is supported so far in aarch64" would be needed.
  * Or a change so that `-mabi=ilp32` / `-mabi=lp64` is selected depending on `CONFIG_64BIT` (and whatever else would be needed)

File: arch/posix/CMakeLists.txt
Comment (aescolar) [reply arch/posix/CMakeLists.txt:9]: & ending it with an else() with a warning "unknown architecture, couldn't select word-size specific compile options, expect problems" would be nice. (But I'd expect native_posix (not 64) to compile without problems in an x86 linux with the -m32 as is here)

**Review by aescolar — DISMISSED** [review]
Thanks @cfriedt , a few requests below

**Comment by cfriedt** [comment]
Thanks for the reviews - I'll make the requested changes today

File: arch/posix/CMakeLists.txt
Comment (cfriedt) [reply arch/posix/CMakeLists.txt:6]: It's a bit tricky. Since Aarch64 is not multilib capable, it doesn't really make sense to specify the abi for the native compiler.
However, `-fPIC` does work regardless of whether the system is running in 32-bit mode or 64-bit mode and it is required for 64-bit mode.

File: arch/posix/CMakeLists.txt
Comment (aescolar) [inline arch/posix/CMakeLists.txt:11]: I cannot test it, but I really doubt this will work for native_poxis(_32) as is. So I'd add a if !CONFIG_64BIT => error

**Review by aescolar — COMMENTED** [review]
I would still like the if for 64 vs 32 bit in aarch64, but as it is now I can already remove my -1 and leave it at +-0 , as it does not worsen anything and improves things a bit.
Though I'd hope for  @tejlmand 's feedback

**Comment by cfriedt** [comment]
@aescolar - you made a really good point about address sizes. Maybe that's justification enough to use CONFIG_64BIT.

That leaves us with

Option 3:

Your original suggestion of making the abi match CONFIG_64BIT.

My next question: How do I make that the default for Aarch64?

**Comment by aescolar** [comment]
> My next question: How do I set a default depending on the userspace wordsize?

I would not know how to do that without user involvement (meaning the user selecting the appropriate native_posix or native_posix_64 target)
I guess I see these ways forward:
* If you don't want to spend a lot of time with this, we can just add a warning in the `if "aarch64"` saying something like "support for native_posix for in ARM is very preliminary, use at your own risk, patches welcome", and a comment telling people to ensure they target native_posix_64 if their userspace is 64 bit and native_posix if it is 32.
* (assuming the user has multilib support in his distro) Assuming that in aarch64 all that is needed to switch between them is `-mabi=ilp32` / `-mabi=lp64`, select them based on CONFIG_64BIT (just like for x86_64 with `-m32` and `-m64`)
* If you can find a nice way of detecting if the user has a given userspace (or both), then when CONFIG_64 was selected or not, set the compile flags appropriately, (and if not supported error out: So say the user is targeting native_posix(_32) (meaning CONFIG_64BIT=n) and there is no user space for it => error in cmake )

**Review by cfriedt — COMMENTED** [review]
> > My next question: How do I set a default depending on the userspace wordsize?
> * If you can find a nice way of detecting if the user has a given userspace (or both), then when CONFIG_64 was selected or not, set the compile flags appropriately, (and if not supported error out: So say the user is targeting native_posix(_32) (meaning CONFIG_64BIT=n) and there is no user space for it => error in cmake )

I think I like the last option because it would Just Work (TM) or error-out. I'm curious though - is `CONFIG_64BIT=y|n` an option that is explicitly enabled / disabled for complete coverage? Would failure in either of those cases prevent automated builds from succeeding?

The following python one-liner does the trick for reporting what the size of a pointer is.

```
python -c 'import struct; print(struct.calcsize("P")*8)'
```

File: arch/posix/CMakeLists.txt
Comment (aescolar) [reply arch/posix/CMakeLists.txt]: Uhm..
One clarification: CONFIG_64BIT is set when you select the target (board) you build for. So if you `-DBOARD=native_posix_64` you are asking to build targeting native_posix_64 which sets CONFIG_64BIT (just like when you select a RISCV-64 based board. Note that native_posix* is the only not-crosscompiled target)
If you target native_posix you are explicitly asking for the 32 bit version.
So, in that regard, if somebody asks for native_posix_64 and you decide for them to build the 32bit version (because that's the only user space they have) you would be "cheating", and probably confusing them. In that case, and if you wanted to be extra nice, I would just error out with a nice explanation ("You can't build native_posix_64 in a 32bit OS")

> Option 1:
> CONFIG_64BIT is not set by default on y machine, and the build fails (as it should?)

=> The fault in this case is in selecting native_posix(_32).

> Option 2:
> Aarch64 requires -fPIC but it also does not hurt to use it for Aarch32. Use it as a blanket option?

That should be fine. I do not have any quarrel with setting this option always :)

> I guess the underlying question is: is it "bad" to run native_posix tests on aarch64 without caring if it is 32-bit or 64-bit?

Not worse than in x86: It should just not work: if you try to build for one of them, it won't if you do not have the libC for that one (32/64), or fail to run later if the loader cannot handle it.

**Review by aescolar — COMMENTED** [review]
(The rabbit hole is quite deep :) )
Thanks for the improvements, and for all the extra knowledge you are bringing, a few extra comments

File: arch/posix/CMakeLists.txt
Comment (aescolar) [reply arch/posix/CMakeLists.txt]: Note that the only proper way to support both native_posix(_32) and native_posix_64 is having multilib support in your host. If you don't, then you can only use one. And that is probably just fine for whoever is trying to build native_posix in ARM.

**Review by aescolar — CHANGES_REQUESTED** [review]
How this PR is right now I cannot approve it (or merge it).
Now this is overriding from cmake a kconfig option and user explicit choice. Meaning, this is assuming it knows what the user wants, and that the user is making a mistake and overriding it. That is not ok.

File: doc/getting_started/toolchain_3rd_party_x_compilers.rst
Comment (tejlmand) [inline doc/getting_started/toolchain_3rd_party_x_compilers.rst:27]: Remove, this is already documented here: https://docs.zephyrproject.org/latest/getting_started/toolchain_other_x_compilers.html#other-cross-compilers

File: arch/posix/CMakeLists.txt
Comment (tejlmand) [inline arch/posix/CMakeLists.txt:34]: error out also means there is no need to pollute the CMake cache with this variable, which is actually a Kconfig setting.

File: arch/posix/CMakeLists.txt
Comment (tejlmand) [reply arch/posix/CMakeLists.txt:45]: Why not ?
He may still be on 32bit userspace, and having `CONFIG_64BIT=n`, in which case everything is fine, and `-fPIC` below will not be added as compile option.

Although I agree he could move the `-fPIC` code line below  into the previous `if (CONFIG_64BIT)` body.

File: arch/posix/CMakeLists.txt
Comment (tejlmand) [reply arch/posix/CMakeLists.txt:46]: which is probably why he safeguarded the compile option with `if (CONFIG_64BIT)`

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please remove the duplicated docs.

File: arch/posix/CMakeLists.txt
Comment (aescolar) [reply arch/posix/CMakeLists.txt:45]: You can ignore this particular comment of mine, not worthwhile.

File: arch/posix/CMakeLists.txt
Comment (aescolar) [reply arch/posix/CMakeLists.txt:46]: This comment was related to a completely different content of this PR around this line, and is not applicable anymore as the PR is now.

Comment (tejlmand) [reply arch/posix/CMakeLists.txt:46]: got it.

File: doc/getting_started/toolchain_3rd_party_x_compilers.rst
Comment (cfriedt) [reply doc/getting_started/toolchain_3rd_party_x_compilers.rst:27]: Sure. I was actually asked to add it in this very PR, but will do.

File: arch/posix/CMakeLists.txt
Comment (cfriedt) [reply arch/posix/CMakeLists.txt:32]: OK, so then is the concensus is to keep the script that checks what the word-size of userspace is and error-out if it does not match the value of `CONFIG_64BIT`?

**Review by aescolar — DISMISSED** [review]

**Comment by cfriedt** [comment]
@tejlmand - duplicate docs removed. Can you please unblock?

File: arch/posix/CMakeLists.txt
Comment (tejlmand) [inline arch/posix/CMakeLists.txt:10]: This will not work on windows.
Is that acceptable ?
After all we are using a cross compiler, so I assume it should be possible to build on windows.

**Review by tejlmand — CHANGES_REQUESTED** [review]
~Is it acceptable to block this from cross-compiling on windows ?~

Edit: I just noticed, you are actually not cross compiling, you are just setting `ZEPHYR_TOOLCHAIN_VARIANT=cross-compile`.

Got tricked by:
```
export ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
export CROSS_COMPILE=/usr/bin/arm-none-eabi-
```

But still a valid question, if we should go down a path where we use commands not available on all platforms supported.

File: arch/posix/CMakeLists.txt
Comment (cfriedt) [reply arch/posix/CMakeLists.txt:10]: Is `native_posix` supported under windows to begin with?

Comment (cfriedt) [reply arch/posix/CMakeLists.txt:10]: Also, on the machine I was testing on, I originally was cross-compiling from aarch64-linux-gnu to arm-non-eabi :-) I know it sounds masochistic but really it's not that bad.

Comment (aescolar) [reply arch/posix/CMakeLists.txt:10]: @tejlmand native_posix is meant to be used to compile targeting your host, so you can test and debug in your own workstation. It uses the host libC, and the host kernel APIs (it requires the kernel offering POSIX threads and other POSIX goodies).
Today I do not think it is really possible to crosscompile from Windows, and would not be too useful to do so (It is a quite bizarre use case, like people developing Linux apps in Windows, in which case they probably end up using WSL or so, which should then work).
Crosscompiling from one Linux platform to another would still be possible, but I would see it as a niche (e.g. using cheaper HW for multidevice tests yet wanting a Linux host below Zephyr).
In principle the POSIX arch can target any POSIX compatible OS, but in reality it is only working on Linux so far (and so far only x86).
Running natively on the Windows kernel is not possible at all or intended by design.

Comment (tejlmand) [reply arch/posix/CMakeLists.txt:10]: @aescolar I know the target `native_posix` is for local host compilation (and I'm not sure that will work on windows in first place, unless you use WSL, cygwin, or similar), I just got tricked by @cfriedt setting cross compiler at first (which is why I edited the comment).

But the question still remains valid, do we want to call out to host commands that may not be available on a system that we are supporting without some safeguard ?

People tend to copy-paste, and suddenly such code could show up in other parts of the build system, breaking it for windows users.
Therefore I would prefer if there were a generic way for doing this test, or ensure a `find_program()` is used, so that proper error message can be printed to user if running on a system where `arch` for some reason is not available.

Comment (aescolar) [reply arch/posix/CMakeLists.txt:10]: Fair enough. I guess it could try call arch and if it cant throw a warning but assume x86_64 which is the most typical case

Comment (tejlmand) [reply arch/posix/CMakeLists.txt:10]: as minimum:
```
find_program(ARCH_EXECUTABLE arch)

if(${ARCH_EXECUTABLE} STREQUAL ARCH_EXECUTABLE-NOTFOUND)
  # Error out or warning ?
else()
  execute_process(
    COMMAND
    ${ARCH_EXECUTABLE}
    OUTPUT_VARIABLE
    BUILD_ARCH
    OUTPUT_STRIP_TRAILING_WHITESPACE
  )
endif()
```

**Review by tejlmand — CHANGES_REQUESTED** [review]
See discussion: https://github.com/zephyrproject-rtos/zephyr/pull/24843#discussion_r421319079

File: arch/posix/CMakeLists.txt
Comment (cfriedt) [reply arch/posix/CMakeLists.txt:10]: If it's easier to digest, we can use the POSIX-required `uname -m`.

Comment (tejlmand) [reply arch/posix/CMakeLists.txt:10]: > Executable should exist on each platform that is POSIX-compiliant.

which windows is not.

As the Zephyr build system is also supporting windows, then there are a couple of reason why calls to host tools should be done with care.
- The tool may not be available, which is why `find_program` is used, and that goes in general.
  `west`, `dtc`, `c compiler`, etc. are all looked up using `find_program`
- Users may see this code, and consider this a nice way to do such testing, and copy it into other parts of the build system, and then refer: `But it's done like that in xyz.cmake`.

The support for multiple OS'es is also why `python` is used, and not `sed`, `awk`, and similar tools.

And then finally for my own sake.
Whenever I review CMake code, I prefer not to consider whether this part of CMake code might be executed on windows or not.

Comment (tejlmand) [reply arch/posix/CMakeLists.txt:10]: A possibility would also be to extend your python script:
`python -c 'import platform; print(platform.machine())'
returns `x86_64` on Linux, AMD64 on windows 64 bit.

Unsure what 32 bit systems return as the Python docs are a bit brief on this: https://docs.python.org/3.8/library/platform.html#platform.machine

but for this small task, using `find_program(ARCH ...)` and then `if(${ARCH_EXECUTABLE} STREQUAL ARCH_EXECUTABLE-NOTFOUND)` the simplest approach.

Comment (cfriedt) [reply arch/posix/CMakeLists.txt:10]: Nice - that works really well. Also, found `CMAKE_HOST_SYSTEM_NAME`. 

I wonder if it would make sense to split things out into `CMAKE_HOST_SYSTEM_NAME`.`CMAKE_HOST_SYSTEM_PROCESSOR`.cmake and then just include the one file (if it exists? is that possible?).

I'm not a CMake expert by any means, so thanks for the tips :-)

File: arch/posix/CMakeLists.txt
Comment (aescolar) [inline arch/posix/CMakeLists.txt:3]: Minor: OS & ARCH are very ambiguous in this context.

I would change this comments to say something like:
"Set necessary compiler & linker options for this host OS"
"Set necessary compiler & linker options for this host architecture"
"Set necessary compiler & linker options for this specific host OS & architecture"

File: arch/posix/Linux.aarch64.cmake
Comment (aescolar) [inline arch/posix/Linux.aarch64.cmake:20]: This recommendation is not too appropriate. 2 reasons:
a) if they are compiling for native_posix, it would be better to ask them to select the target board (here `-DBOARD=native_posix_64`) instead of changing the underlying kconfig options
b) But anyhow, this is an ARCH cmake file (shared between several boards), so you do not know if they are targeting native_posix at all.
=> Maybe better change the comment to 
"If you were targeting native_posix_64, target native_posix instead"
and in the msg below:
"If you were targeting native_posix, target native_posix_64 instead"

**Review by aescolar — COMMENTED** [review]
Nothing too bad from my point of view but worthwhile changing

File: arch/posix/CMakeLists.txt
Comment (tejlmand) [inline arch/posix/CMakeLists.txt:10]: This is really changing current behavior.
Have you ensured this will work as expected on all platforms / boards ?
One thing is to ensure we are not breaking officially supported platforms, but there are still platforms / toolchains which are not officially supported, but still works, and people are using.

Also, this means each platform needs a file, even if the content is identical to another systems file.
But it is also nice to split into logical groups, so maybe something like:
```suggestion
if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/${CMAKE_HOST_SYSTEM_NAME}.${CMAKE_HOST_SYSTEM_PROCESSOR}.cmake)
  # @Intent: Set necessary compiler & linker options for the specific OS-ARCH tuple
  include(${CMAKE_HOST_SYSTEM_NAME}.${CMAKE_HOST_SYSTEM_PROCESSOR}.cmake)
else()
  if (CONFIG_64BIT)
    # some gcc versions fail to build without -fPIC
    zephyr_compile_options(-m64 -fPIC)
    zephyr_link_libraries(-m64)
  else ()
    zephyr_compile_options(-m32)
    zephyr_link_libraries(-m32)
  endif ()
endif()
```

that way we ensure current behavior is not changing, while allowing the flexibility of placing platform specific files.
Let's wait with extending with `${CMAKE_HOST_SYSTEM_PROCESSOR}.cmake` and `${CMAKE_HOST_SYSTEM_NAME}.cmake` until needed.
We should in general try to minimize host dependent variations, although it is needed in some case, like this PR shows.

File: arch/posix/x86_64.cmake
Comment (tejlmand) [inline arch/posix/x86_64.cmake:1]: This file can be removed, see other comment.

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: arch/posix/Linux.aarch64.cmake
Comment (cfriedt) [reply arch/posix/Linux.aarch64.cmake:20]: I see - for some reason I was under the impression that arch/posix was specifically for native_posix. Personally, that's actually all I'm concerned with atm.

Possibly the better solution is to put my changes into `boards/posix/native_posix/` rather than `arch/posix/`, and then just surround the "-m32 / -m64" stuff with a check for `x86_64`. Are there any other arches for `arch/posix` that use "-m32 / -m64"?

File: arch/posix/CMakeLists.txt
Comment (cfriedt) [reply arch/posix/CMakeLists.txt:10]: See previous [comment](https://github.com/zephyrproject-rtos/zephyr/pull/24843#discussion_r422088474)

File: arch/posix/Linux.aarch64.cmake
Comment (aescolar) [reply arch/posix/Linux.aarch64.cmake:20]: Doing it here is correct. We just need to write a message that does not confuse users too much.

>Are there any other arches for arch/posix

You mean boards? In the tree only native_posix_64 sets CONFIG_64BIT by default. But there is off-tree boards and anyhow users are free to set it differently.

File: arch/posix/CMakeLists.txt
Comment (cfriedt) [reply arch/posix/CMakeLists.txt]: Done!

**Comment by aescolar** [comment]
@cfriedt => slack

File: arch/posix/Linux.aarch64.cmake
Comment (cfriedt) [reply arch/posix/Linux.aarch64.cmake]: @aescolar - I have to remember to get your feedback before respinning the PR ;-) Your feedback is very good, btw.

**Comment by cfriedt** [comment]
Thanks @aescolar and @tejlmand for your dialog and assistance shepherding this one in.

**Comment by carlescufi** [comment]
@tejlmand and @stephanosio any chance of getting this reviewed today?

**Comment by carlescufi** [comment]
@cfriedt can you fix Gitlint please?

**Review by aescolar — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Comment by cfriedt** [comment]
Can anyone merge?

**Pull request merged**
